src/Loader/FileLoader.php

```diff
@@ -5,29 +5,30 @@
 namespace AutoMapper\Loader;
 
 use AutoMapper\Generator\MapperGenerator;
-use AutoMapper\MapperGeneratorMetadataInterface;
+use AutoMapper\MapperMetadata\MapperGeneratorMetadataInterface;
 use PhpParser\PrettyPrinter\Standard;
 use PhpParser\PrettyPrinterAbstract;
 
 /**
  * Use file system to load mapper, and persist them using a registry.
  *
  * @author Joel Wurtz <jwurtz@jolicode.com>
+ *
+ * @internal
  */
 final class FileLoader implements ClassLoaderInterface
 {
     private readonly PrettyPrinterAbstract $printer;
     private ?array $registry = null;
 
     public function __construct(
-        private readonly MapperGenerator $generator,
         private readonly string $directory,
         private readonly bool $hotReload = true,
     ) {
         $this->printer = new Standard();
     }
 
-    public function loadClass(MapperGeneratorMetadataInterface $mapperMetadata): void
+    public function loadClass(MapperGenerator $mapperGenerator, MapperGeneratorMetadataInterface $mapperMetadata): void
```

Comment:
I refactored this method, in order to remove the `MapperGenerator` from `EvalLoader` / `FileLoader`'s constructor.
This is really easier to refactor then, because, we don't need to create a full `MapperGenerator` [in the tests](https://github.com/nikophil/automapper/blob/feat/deffer-resolve-transformer/tests/AutoMapperBaseTest.php#L36)